tests/helpers/test_script.py

```diff
@@ -1400,7 +1399,8 @@ async def test_wait_template_with_utcnow_no_match(hass: HomeAssistant) -> None:
     assert_action_trace(
         {
             "0": [{"result": {"wait": {"completed": False, "remaining": None}}}],
-        }
+        },
+        expected_script_execution="cancelled",
```

Comment:
Stopping a script is not cancelling it. It could be an intended stop.

../Frenck

Reply:
My initial idea would be to add "stopped" but when I looked in the frontend, this text was at "cancelled":
```

    // The exception is in the trace itself or in the last element of the trace
    // Script execution stopped by async_stop called on the script run because home assistant is shutting down, script mode is SCRIPT_MODE_RESTART etc:
    | "cancelled";
```

Reply:
FWIW It looks like we also set `cancelled` above on line 418 when stop is set causing it to return from that function so it may be wrong in a few places.

Reply:
I'm making an adjustment to return values when cancelled is raised here: https://github.com/home-assistant/core/pull/98501 -- another angle for looking at the same issue.

Reply:
Reading the code, the only time "cancelled" is used is when "stop" is called. While cancellation seems to be used internally as part of stopping, i'm not sure I see another use of cancellation anywhere.

There is only one use of "cancelled" which is one out of 3 or 4 cases stop is called.

WDYT about:
- introduce a new status `stopped`
- everywhere stop is set, the status "stopped" is set
- Depending on the context, may be able to add additional information to the trace where that stop is called (e.g. it can include the detail that it was stopped because mode: `restart`, or it can do nothing if stop was called externally)
- add a new frontend code "stopped" as well etc
- deprecated cancelled since it seems to be an internal detail

Another option could be to introduce other modes like `failed_restart` that gives a more specific error message for cases we know about where the script is stopped because another automation ran and stopped it.

Not a script expert, but intent is to keep the conversation moving.

Reply:
My original idea was indeed to introduce stopped, but then I came across that comment. I think using stopped can be something good.

This is code I never touched so I don't know the ins and outs, but I think having a way to know why something happens would be usefull as well. The bug that made me make this PR, is that the user got a vague error. Even when this is merged, the user can see that it's stopped or cancelled, but still doesn't know why. (Maybe they accidentally left it in a scene? Maybe they are triggering another automation that disables the current one?) Can we find a way to fix this and add room for extending this in the future?